### PR TITLE
[Harness] Add an extra index that will use full uris.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Vsdrops does not support serving a static html. Therefore we need to use
full uris that will be used to download the logs. To make things less
dangerous, we leave the xamarin-storage report as it was and create a
new one for vsdrops.

This means that:

1. xamarin-storage index.html is left as is.
2. vsdrops_index.html contains full uris to download (the env var will
   have to be set in the step) and js and css are in the header.
3. because we use and env var, jenkins won't generate the
   vsdrops_index.html only device pipelines will.

For this to take effect needs updates in the device pipelines. The
solution is not yet optimal since we need to add some workaround to
rather than make the monitoring person open a text file, we should
display it in the browser.

### CODE REVIEW

File: tests/xharness/Jenkins/Reports/HtmlReportWriter.cs [hunk 1]
```diff
@@ -17,20 +18,66 @@ class HtmlReportWriter : IReportWriter {
...
+						writer.WriteLine (line);
+					}
+				}
+				writer.WriteLine("</script>");
+			} else {
+				writer.WriteLine (@"<script type='text/javascript' src='xharness.js'></script>");
+			}
+		}
+
+		void IncludeCSS (StreamWriter writer)
```
Comment (spouliot) [inline tests/xharness/Jenkins/Reports/HtmlReportWriter.cs:65]: minor: should be `Css`

File: tests/xharness/Jenkins/Jenkins.cs [hunk 2]
```diff
@@ -307,7 +311,9 @@ public void GenerateReport ()
 			try {
 				lock (report_lock) {
 					var report = Path.Combine (LogDirectory, "index.html");
+					var vsdropsReport = Path.Combine (LogDirectory, "vsdrops_index.html");
 					var tmpreport = Path.Combine (LogDirectory, $"index-{Helpers.Timestamp}.tmp.html");
+					var tmpVsdropsReport= Path.Combine (LogDirectory, $"vsdrops_index-{Helpers.Timestamp}.tmp.html");
```
Comment (chamons) [inline tests/xharness/Jenkins/Jenkins.cs:316]:
```suggestion
					var tmpVsdropsReport = Path.Combine (LogDirectory, $"vsdrops_index-{Helpers.Timestamp}.tmp.html");
```

File: tests/xharness/Jenkins/Reports/HtmlReportWriter.cs [hunk 3]
```diff
@@ -17,20 +18,66 @@ class HtmlReportWriter : IReportWriter {
 		readonly Jenkins jenkins;
 		readonly IResourceManager resourceManager;
 		readonly IResultParser resultParser;
+		private readonly string? linksPrefix;
```
Comment (chamons) [inline tests/xharness/Jenkins/Reports/HtmlReportWriter.cs:21]: Isn't private by default, I think the private prefix is unneeded.

File: tests/xharness/Jenkins/Reports/HtmlReportWriter.cs [hunk 4]
```diff
@@ -17,20 +18,66 @@ class HtmlReportWriter : IReportWriter {
...
+			this.linksPrefix = linksPrefix;
+			this.embededResources = embeddedResources;
 		}
 
+		string GetLinkPath (string path)
+			=> linksPrefix != null ? Path.Combine (linksPrefix, path) : path;
+
+		string GetResourcePath (string resource)
+		{
+			var executingDir= Path.GetDirectoryName (Assembly.GetExecutingAssembly ().Location);
```
Comment (chamons) [inline tests/xharness/Jenkins/Reports/HtmlReportWriter.cs:44]:
```suggestion
			var executingDir = Path.GetDirectoryName (Assembly.GetExecutingAssembly ().Location);
```

File: tests/xharness/Jenkins/Reports/HtmlReportWriter.cs (hunk 3 above)
Comment (mandel-macaque) [inline tests/xharness/Jenkins/Reports/HtmlReportWriter.cs:21]:
```suggestion
		readonly string? linksPrefix;
```

File: tests/xharness/Jenkins/Reports/HtmlReportWriter.cs [hunk 5]
```diff
@@ -17,20 +18,66 @@ class HtmlReportWriter : IReportWriter {
 		readonly Jenkins jenkins;
 		readonly IResourceManager resourceManager;
 		readonly IResultParser resultParser;
+		readonly string? linksPrefix;
+		private readonly bool embededResources;
```
Comment (mandel-macaque) [inline tests/xharness/Jenkins/Reports/HtmlReportWriter.cs:22]:
```suggestion
		readonly bool embededResources;
```

**Pull request merged**